core/runner.py

```diff
@@ -39,18 +42,23 @@ class DMLRunner(object):
     def __init__(self, config_manager):
         """
         Sets up the unique identifier of the DML Runner and the local dataset path.
+        
+        Sets up IPFS client for _communicate.
         """
         config = config_manager.get_config()
         self.iden = str(uuid.uuid4())[:8]
-        
         self.config = dict(config.items("RUNNER"))
+        self.client = ipfsapi.connect(
+            config.get('BLOCKCHAIN', 'host'), 
+            config.getint('BLOCKCHAIN', 'ipfs_port'))
+        self.port = config.getint('BLOCKCHAIN', 'http_port')
```

Comment:
In #33 you set up an IPFS Client in DatasetManager, and now you're setting one up here (they seem to be identical). Not sure if I missed this convo but was this intentional? If not, I think it would be better to set up the client in the Bootstrapper and pass it to the Dataset Manager and the Communication Manager -> Optimizer on instantiation.

Reply:
Yeah this was intentional. But I can do the other thing as well.

Reply:
Oh wait yeah the reason I did this instead of Bootstrapper is because if I do it with Bootstrapper only then the Runner/Dataset Manager wouldn't pass any tests unless they were bootstrapped. So I'm actually going to leave it like this for time being.

Reply:
Okee.

Reply:
Done